validator/remote/src/main/java/tech/pegasys/teku/validator/remote/eventsource/EventSourceBeaconChainEventAdapter.java

```diff
@@ -181,6 +182,12 @@ private synchronized boolean switchToFailoverEventStreamIfAvailable() {
     if (failoverBeaconNodeApis.isEmpty()) {
       return false;
     }
+    // No need to change anything if current node is READY
+    if (beaconNodeReadinessManager
+        .getReadinessStatus(currentBeaconNodeUsedForEventStreaming)
+        .equals(READY)) {
```

Comment:
IMU the idea here is to check if the current BN is ready before switching to another one, right?
If that's the case, shouldn't we rather check `isReady()` instead of `equals(READY)`?
What if we refactor the `findReadyFailoverAndSwitch` instead and add the `currentBeaconNodeUsedForEventStreaming` to the stream and let it select the best available BN?

Reply:
> shouldn't we rather check isReady()
It will remove an option to switch from optimistic_ready to ready

> What if we refactor the findReadyFailoverAndSwitch instead 
It will not guarantee order on each call. What I've tried  to do - to keep the current node if switch is not needed. This method could be called even in case Primary, ready, active, Failover - callback on disconnect and it will change primary to failover in this case.